src/realm/db.cpp

```diff
@@ -1643,6 +1684,8 @@ void DB::wait_for_change_release()
 
 void DB::enable_wait_for_change()
 {
+    if (m_fake_read_lock_if_immutable)
+        return;
```

Comment:
Should this be an error instead?

Reply:
If you allow this, then you should probably allow calling `wait_for_change` too.

Reply:
I'll disallow instead.